### DISCUSSION TITLE
Dependencies: extra requirements need to be forwarded to kombu ?

### DISCUSSION THREAD

**Discussion by NielkS** [celery/celery#7088] [Q&A]
Hello, I'm investigating a dependency issue I had after the the minimal version of redis was updated as part of kombu 5.2.0 (ref: https://github.com/celery/kombu/issues/1419#issuecomment-968064714 )

Given the following requirements:
```
celery[redis]==5.1.0
redis==3.3.11
```
I would expect kombu==5.2.0 to be installed, not kombu==5.2.1 -- _that would still not work, because kombu 5.2.0 does requires redis>=3.4.1, but that is the point of kombu ticket mentioned above_

That is not what happens, though, kombu==5.2.1 is installed when I try this.
I think this would be solved by adding kombu[redis] in [requirements/extras/redis.txt](https://github.com/celery/celery/blob/master/requirements/extras/redis.txt) - there is a similar pattern in https://github.com/celery/celery/blob/master/requirements/extras/sqs.txt containing `kombu[sqs]`.

Should I create a ticket and add `kombu[xxx]` for all celery extra requirements with a corresponding kombu extra requirement ?

**Comment by thedrow** [comment]
I agree that intuitively they should be forwarded to kombu but how would you propose to do that?
Also, what if Celery uses a client library for a result backend and we don't care about the version kombu requires since that one is for the message broker.

I think we should split up the packages instead which would make it easier to enforce these constraints.

**Comment by auvipy** [comment]
also https://github.com/celery/kombu/issues/1419#issuecomment-968113102

**Reply by auvipy** [reply]
And https://github.com/celery/celery/pull/7069/files
